src/CommunityToolkit.Maui.Core/Essentials/FileSaver/FileSaverImplementation.android.cs

```diff
@@ -14,26 +15,32 @@ public sealed partial class FileSaverImplementation : IFileSaver
 {
 	static async Task<string> InternalSaveAsync(string initialPath, string fileName, Stream stream, CancellationToken cancellationToken)
 	{
-		var status = await Permissions.RequestAsync<Permissions.StorageWrite>().WaitAsync(cancellationToken).ConfigureAwait(false);
-		if (status is not PermissionStatus.Granted)
+		const string externalStorageBaseUrl = "content://com.android.externalstorage.documents/document/primary%3A";
+
+		AndroidUri? filePath = null;
+
+		if (!OperatingSystem.IsAndroidVersionAtLeast(33))
 		{
-			throw new PermissionException("Storage permission is not granted.");
+			var status = await Permissions.RequestAsync<Permissions.StorageWrite>().WaitAsync(cancellationToken).ConfigureAwait(false);
+			if (status is not PermissionStatus.Granted)
+			{
+				throw new PermissionException("Storage permission is not granted.");
+			}
 		}
 
-		const string baseUrl = "content://com.android.externalstorage.documents/document/primary%3A";
 		if (Android.OS.Environment.ExternalStorageDirectory is not null)
 		{
 			initialPath = initialPath.Replace(Android.OS.Environment.ExternalStorageDirectory.AbsolutePath, string.Empty, StringComparison.InvariantCulture);
 		}
 
-		var initialFolderUri = AndroidUri.Parse(baseUrl + HttpUtility.UrlEncode(initialPath));
+		var initialFolderUri = AndroidUri.Parse(externalStorageBaseUrl + HttpUtility.UrlEncode(initialPath));
 		var intent = new Intent(Intent.ActionCreateDocument);
 
 		intent.AddCategory(Intent.CategoryOpenable);
 		intent.SetType(MimeTypeMap.Singleton?.GetMimeTypeFromExtension(MimeTypeMap.GetFileExtensionFromUrl(fileName)) ?? "*/*");
 		intent.PutExtra(Intent.ExtraTitle, fileName);
 		intent.PutExtra(DocumentsContract.ExtraInitialUri, initialFolderUri);
```

Comment:
`DocumentsContract.ExtraInitialUri` is only supported in Android API 26+. We support Android 21+ because .NET MAUI supports Android 21+. 

Could we add a branch to support Android API 21-25?

I know this bug isn't introduced in this PR, but any chance that it's a quick/easy fix we could squeeze in here?

Reply:
Hmm, I don’t see the warning that we use unsupported API here, but if it doesn’t work on old api, it would be nice to fix.

Reply:
I'll have a look on that

Reply:
After some testing it seams that EXTRA_INITIAL_URI is simply not considered by API Level 21 - 25. It is added by `Intent.PutExtra(string name, string data)` so it is not giving a warning since there is no check on the string. 
There is no chance to implement this in API level below 26 since Intent.ActionOpenDocumentTree is a build-in Intent of the AndroidSDK.

The only thing we could do is set a minimal api level on `InternalPickAsync(string initialPath)` and add a wrapper method without `initialPath` without warning.

Please let me know if this would make sense.

Reply:
Yup - that makes perfect sense. Thanks for digging into it! 

Could you add the following attribute to all public methods that use this API?
`[SupportedOSPlatform("Android26.0")]`

Here's an example:

https://github.com/CommunityToolkit/Maui/blob/0ab0fe22f4e701d1f463e243f85c2ddb07354a6a/src/CommunityToolkit.Maui.Core/Platform/StatusBar/StatusBar.android.cs#L10

If it is a public method in cross-platform code, then we'll also need to add a similar attribute covering all iOS, macOS, Windows and Tizen versions, like this:

```cs

[SupportedOSPlatform("Android26.0")]
[SupportedOSPlatform("iOS")]
[SupportedOSPlatform("macOS")]
[SupportedOSPlatform("Tizen")]
[SupportedOSPlatform("Windows")]

```

Reply:
And let's implement a better error message too. 

We just put in a `if` statement to check the current version of Android and then `throw new NotSupportedException("FilePicker is only supported on Android API 26 and higher")]`. 

This more descriptive error will better help devs in the future when they see this stack trace.

Reply:
The code will work on api level below 26, it just won't start in the InitialUri, should I still add the attributes? Just to be sure.

Reply:
@redek91, could you please check if code works on Android 26 and below. If Android correctly resolves it and ignores this API, then we shouldn't add this SupportedOSPlatform attribute.
Futhermore, if it doesn't work, just wrap the single line with if (OperationSystem.AndroidVersionAtLeast(26))

Reply:
I checked on an android api level 21 virtual device, the document tree opens on /, and ignores the extra added to the intent action.

Reply:
Let's add Trace.WriteLine("DocumentsContract.ExtraInitialUri is only supported on Android 26 and later"); @brminnick what do you think?

Reply:
> I checked on an android api level 21 virtual device, the document tree opens on /, and ignores the extra added to the intent action.

Ah - ok - so the API is still usable and doesn't crash the app. Good to know! 

Yup - I'm cool with adding `Trace.WriteLine`. Let's make it specific to this use-case since the average dev likely doesn't know that we're using `DocumentsContract.ExtraInitialUri` under the hood: 

```cs
Trace.WriteLine("Specifying a folder path is only supported on Android 26 and later");